fair_drop/suspicious.py

```diff
@@ -0,0 +1,181 @@
+# See https://github.com/convex-labs/honestnft-shenanigans/issues/86 for more details
+# Given a collection of NFTs on OpenSea, detect suspicious NFTs
+import time
+from argparse import ArgumentParser
+import logging
+
+import cloudscraper
+import pandas as pd
+from bs4 import BeautifulSoup
+from requests.adapters import HTTPAdapter, Retry
+
+
+logging.basicConfig(level=logging.INFO)
+
+parser = ArgumentParser()
+parser.add_argument(
+    "-c",
+    "--collection",
+    dest="collection_address",
+    help="Address of the NFT collection",
+    metavar="COLLECTION",
+    required=True,
+)
+parser.add_argument(
+    "-s",
+    "--sleep",
+    dest="sleep_timer",
+    help="Sleep parameter",
+    metavar="SLEEP",
+    required=False,
+    default=30,
+)
+parser.add_argument(
+    "-r",
+    "--retry",
+    dest="retries_on_rate_limit",
+    help="Number of retries to attempt when rate limited by OpenSea",
+    metavar="RETRY",
+    required=False,
+    default=3,
+)
+
+args = parser.parse_args()
+
+COLLECTION_CSV_PATH = f"fair_drop/suspicious_{args.collection_address}.csv"
+
+scraper = cloudscraper.create_scraper()
+# Configration of cloudscraper underlying requests module
+# CloudScraper is a sub-class of Session
+retry_strategy = Retry(
+    total=3,
+    status_forcelist=[429, 500, 502, 503, 504],
+    allowed_methods=["HEAD", "GET", "OPTIONS"],
+    backoff_factor=8,
+    raise_on_status=False,  # If retries fail, return response instead of raising exception
```

Comment:
If you add `respect_retry_after_header=True` , you can probably skip the while-loop on LOC 67-...

Reply:
Ohh that's great. I've completely overlooked this one.
I am not sure though if OpenSea has a header that explains how much time to wait for a retry. I've been trying to get rate limited by opensea to check the response headers, but to no avail :sweat_smile:

Reply:
Naah I've removed that, even wrote a small script to send a bunch of requests at the same time, and couldn't get rate limited

Reply:
Anyway, it's resolved now